core/src/main/scala/io/github/rafafrdz/criteria4s/core/Ref.scala

```diff
@@ -1,43 +1,40 @@
 package io.github.rafafrdz.criteria4s.core
 
-sealed trait Ref[D <: CriteriaTag] {
-  def ref: Criteria[D]
+import io.github.rafafrdz.criteria4s.core.Show.ShowColumn
 
-  override def toString: String = ref.value
+sealed trait Ref[D <: CriteriaTag, +V] {
+  def asString(implicit show: Show[V, D]): String
 }
 
 object Ref {
 
-  trait Value[+V, D <: CriteriaTag] extends Ref[D]
+  trait Value[D <: CriteriaTag, +V]      extends Ref[D, V]
+  trait Collection[D <: CriteriaTag, +V] extends Ref[D, Seq[V]]
+  trait Col[D <: CriteriaTag]            extends Ref[D, Column]
+  trait Bool[D <: CriteriaTag]           extends Value[D, Boolean] with Criteria[D]
+  trait Range[D <: CriteriaTag, +V]      extends Ref[D, (V, V)]
```

Comment:
This is awesome! I really love that, seriously 😀

I have a question just to really understand the reason. Why did yo switch the position of the Value type constructors? Was it just for aesthetics? in order to fix the same position the same constructor (always the D on the left and the rest on the right? or was it because of some Variance/Covariance position reason?